src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -80,7 +119,15 @@ const std::shared_ptr<AbstractLQPNode>& SQLPipelineStatement::get_unoptimized_lo
     return _unoptimized_logical_plan;
   }
 
-  const auto& parsed_sql = get_parsed_sql_statement();
+  auto parsed_sql = get_parsed_sql_statement();
+
+  const auto* statement = parsed_sql->getStatement(0);
+  if (const auto prepared_statement = dynamic_cast<const hsql::PrepareStatement*>(statement)) {
+    Assert(_prepared_statements, "Cannot prepare statement without prepared statement cache.");
+    parsed_sql = SQLPipelineStatement(prepared_statement->query).get_parsed_sql_statement();
+    _num_parameters = static_cast<uint16_t>(parsed_sql->parameters().size());
+  }
```

Comment:
Could you please add a comment to this block? If I understand it correctly, it is to "unwrap" the actual SQL statement from the prepare statement?